ocs_ci/ocs/small_file_workload.py

```diff
@@ -1,141 +1,210 @@
+                self.crd_data["spec"]["elasticsearch"]["server"],
+                self.crd_data["spec"]["elasticsearch"]["port"],
+            )
+            self.crd_data["spec"]["elasticsearch"]["parallel"] = True
+
+        # Saving the Original elastic-search IP and PORT - if defined in yaml
+        self.backup_es = self.crd_data["spec"]["elasticsearch"]
+
+        # Use the internal define elastic-search server in the test - if exist
+        if self.es:
+            self.crd_data["spec"]["elasticsearch"] = {
+                "url": f"http://{self.es.get_ip()}:{self.es.get_port()}",
+                "server": self.es.get_ip(),
+                "port": self.es.get_port(),
+                "parallel": True,
+            }
+        if self.crd_data["spec"]["elasticsearch"] == {}:
+            log.error(
+                "No ElasticSearch server is available. workload can not be execute"
+            )
+            return False
+
+        return True
+
+    def setup_storageclass(self, interface):
+        """
+        Setting up the storageclass parameter in the CRD object
+
+        Args:
+            interface (str): the storage interface
+
+        """
+        if interface == constants.CEPHBLOCKPOOL:
+            storageclass = constants.DEFAULT_STORAGECLASS_RBD
+        else:
+            storageclass = constants.DEFAULT_STORAGECLASS_CEPHFS
+        log.info(f"Using {storageclass} Storageclass")
+        self.crd_data["spec"]["workload"]["args"]["storageclass"] = storageclass
+
+    def setup_test_params(self, file_size, files, threads, samples):
+        """
+        Setting up the parameters for this test
+
+        Args:
+            file_size (int): the file size in KB
+            files (int): number of file to use in the test
+            threads (int): number of threads to use in the test
+            samples (int): number of sample to run the test
+
+        """
+        self.crd_data["spec"]["workload"]["args"]["file_size"] = file_size
+        self.crd_data["spec"]["workload"]["args"]["files"] = files
+        self.crd_data["spec"]["workload"]["args"]["threads"] = threads
+        self.crd_data["spec"]["workload"]["args"]["samples"] = samples
+
+    def setup_vol_size(self, file_size, files, threads):
+        """
+        Calculating the size of the volume that need to be test, it should
+        be at least twice in the size then the size of the files, and at
+        least 100Gi.
+
+        Since the file_size is in Kb and the vol_size need to be in Gb, more
+        calculation is needed.
+
+        Args:
+            file_size (int): the file size in KB
+            files (int): number of file to use in the test
+            threads (int): number of threads to use in the test
+
+        """
+        vol_size = int(files * threads * file_size * 3)
+        vol_size = int(vol_size / constants.GB2KB)
+        if vol_size < 100:
+            vol_size = 100
```

Comment:
Hope this won't be a problem, i.e. If volume size is 50G and if we write more than 50G then there will be failure

Reply:
the test is plan to write not more then 100 GiB

Reply:
i add a check to verify that we have enough storage capacity for the test.